### ISSUE TITLE
tf.keras compatibility: convert multi-label-classification to Keras Core

### ISSUE THREAD

**Issue by Akorex** [keras-team/keras-core#539]
for tf.keras backwards compatibility

In my attempt to port [Large-scale multi-label classification](https://keras.io/examples/nlp/multi_label_classification) to `keras_core` I hit an issue with the Inference Section. The example from keras.io created a new Inference model after training to include the TextVectorization preprocessing layer so inference can be done by passing strings directly to the model. 

```model_for_inference = keras.Sequential([text_vectorizer, shallow_mlp_model])```

This didn't work for keras_core imports and the attempt to port the example didn't include the particular step from the example. 

PR: #538 

Colab Notebook: https://colab.research.google.com/drive/1kR7W4BJhtDlA5Jx_e4EwUM2FnzKP-D58#scrollTo=xmf6rTLL0G9H

**Comment by grasskin** [comment]
Thank you for opening this issue! I'm getting access denied on the colab notebook, could you change the permissions/send a different link?

**Comment by Akorex** [comment]
I have updated the permissions to the notebook and you should have access to it now.

**Comment by fchollet** [comment]
The notebook you shared doesn't seem to include the line `model_for_inference = ...`. Can you post a shorter, more focused reproduction notebook?

**Comment by Akorex** [comment]
I have added the line `model_for_inference = ...` in the new notebook.  However, I hit an issue with `layers.StringLookup()` using the keras_code import which didn't exist with the tf.keras import

Notebook : https://colab.research.google.com/drive/134sncHaoAaV9xeV7fOHVDZMVu12rAUVI#scrollTo=iwVJj6dK0G9N
